ADApp/pluginTests/test_NDPluginOverlay.cpp

```diff
@@ -0,0 +1,255 @@
+
+    // Test a "normal" case
+    overlayTempCaseStr test1 = {0, 500, 500, 50, 50, 1, 1, 0, 255, 0, NDOverlayCross, NDOverlaySet, 2, {1024, 1024}, NDColorModeMono};
+    appendTestCase(&overlayTestCaseStrs, &test1); 
+    // Test a case with size larger than array
+    overlayTempCaseStr test2 = {0, 500, 500, 5000, 5000, 1, 1, 0, 255, 0, NDOverlayEllipse, NDOverlaySet, 2, {1024, 1024}, NDColorModeMono};
+    appendTestCase(&overlayTestCaseStrs, &test2); 
+    // Test a case with zero size
+    overlayTempCaseStr test3 = {0, 500, 500, 0, 0, 1, 1, 0, 255, 0, NDOverlayRectangle, NDOverlaySet, 2, {1024, 1024}, NDColorModeMono};
+    appendTestCase(&overlayTestCaseStrs, &test3); 
+    // Test a case with negative position
+    overlayTempCaseStr test4 = {0, -500, -500, 50, 50, 1, 1, 0, 255, 0, NDOverlayCross, NDOverlaySet, 2, {1024, 1024}, NDColorModeMono};
+    appendTestCase(&overlayTestCaseStrs, &test4); 
+    // Test a case with position larger than array size
+    overlayTempCaseStr test5 = {0, 1500, 1500, 50, 50, 1, 1, 0, 255, 0, NDOverlayEllipse, NDOverlaySet, 2, {1024, 1024}, NDColorModeMono};
+    appendTestCase(&overlayTestCaseStrs, &test5); 
+    // Test an RGB1 case 
+    overlayTempCaseStr test6 = {0, 500, 500, 50, 50, 1, 1, 0, 255, 0, NDOverlayRectangle, NDOverlaySet, 3, {3, 1024, 1024}, NDColorModeRGB1};
+    appendTestCase(&overlayTestCaseStrs, &test6); 
+    // Test an case with very large width
+    overlayTempCaseStr test7 = {0, 500, 500, 50, 50, 5000, 5000, 0, 255, 0, NDOverlayCross, NDOverlaySet, 2, {1024, 1024}, NDColorModeMono};
+    appendTestCase(&overlayTestCaseStrs, &test7); 
+    // Test an case with zero width
+    overlayTempCaseStr test8 = {0, 500, 500, 50, 50, 0, 0, 0, 255, 0, NDOverlayCross, NDOverlaySet, 2, {1024, 1024}, NDColorModeMono};
+    appendTestCase(&overlayTestCaseStrs, &test8); 
+}
+
+  ~OverlayPluginTestFixture()
+  {
+    delete arrayPool;
+    client.reset();
+    Overlay.reset();
+    driver.reset();
+    //delete downstream_plugin; // TODO: We can't delete a TestingPlugin because it tries to delete an asyn port which doesnt work
+  }
+
+};
+
+BOOST_FIXTURE_TEST_SUITE(OverlayPluginTests, OverlayPluginTestFixture)
+
+
+BOOST_AUTO_TEST_CASE(validate_input_NDArrays)
+{
+  BOOST_MESSAGE("Validating the input NDArrays " <<
+                "- if any of these fail the test-cases are also likely to fail");
+
+  overlayTestCaseStr *pStr = &overlayTestCaseStrs[0];
+  for (size_t i=0; i<overlayTestCaseStrs.size(); i++, pStr++)  {
+    for (size_t dim=0; dim<pStr->arrayDims.size(); dim++) {
+       BOOST_REQUIRE_EQUAL(pStr->arrayDims[dim], pStr->pArrays[0]->dims[dim].size);
+    }
+    BOOST_REQUIRE_EQUAL(pStr->pArrays[0]->dataType, NDFloat32);
+    BOOST_REQUIRE_EQUAL(pStr->arrayDims.size(), pStr->pArrays[0]->ndims);
+  }
+}
+
+
+BOOST_AUTO_TEST_CASE(basic_overlay_operation)
+{
+
+  overlayTestCaseStr *pStr = &overlayTestCaseStrs[0];
+  for (size_t i=0; i<overlayTestCaseStrs.size(); i++, pStr++)  {
```

Comment:
A quick google yields some result: http://www.boost.org/doc/libs/1_62_0/libs/test/doc/html/boost_test/tests_organization/test_cases/test_case_generation.html

I have not tried to use the BOOST_DATA_TEST_CASE yet but at a glance this looks like it might help with the case of testing a set of different input variables. I wouldn't hold up this particular PR with a change like that though!